pkg/filter/transform/transform.go

```diff
@@ -70,7 +71,12 @@ func (f *Filter) Filter(evt event.Event) []event.Event {
 		for idx, elem := range a {
 			subEvt, err := event.New(evt.Context(), elem, event.WithMetadata(evt.Metadata()), event.WithTenant(evt.Tenant()))
```

Comment:
Missed this before, but when you create a subevent like this, the subevent is no longer connected with the parent event, and the ack/nack on the sub event will not be sent to the parent. I think you want to use evt.Clone to create subevent.

Reply:
not sure about that - the sub event is just used to make a partial transformation on the side which will then be injected into the payload of the original event afterwards (we use this for iterating over arrays and apllying a transform iteratively, for example) - so, the original event stays intact and gets forwarded